### ISSUE TITLE
Removing autofocus (and cleanup switching between portrait and landscape formats)

### ISSUE THREAD

**Issue by misc-de** [deltachat/deltachat-desktop#5888]
There are some annoying behaviors when using DeltaChat Desktop on a Linux smartphone. I hope you can fix them :) I'll try my best to explain the behavior, sorry if I haven't managed to do so :P

1. When I start DeltaChat, the focus jumps directly to the "Search" field, triggering the keyboard to appear in the Posh UI (image 1). I first have to tap the (account) navigation on the left to remove the focus from the search field and make the keyboard disappear (image 2).

2. When opening a chat or switching to landscape mode, the input focus jumps directly to the "Message" field and opens the keyboard (image 6). This is rather unsightly and makes using the UI cluttered.

3. When I use the app in portrait mode, I have the (account) navigation on the left and my contacts overview on the right (image 2), which I can select and then use in full-screen chat (image 5) – which is exactly how it should behave.
If I now rotate the smartphone from portrait to landscape mode, I have the (account) navigation on the left, then the contacts, and then the chat view (image 3) – which is perfectly fine initially. However, when I switch back to portrait mode, I'm not back in the overview (navigation and contacts) (image 2), but directly in the first chat of my contacts, as if that had been my active focus. This is certainly due to the autofocus being set to the "message" field.
This is very confusing, as the app rotation also happens when I'm not actively using the app, but rather taking a photo in landscape mode with the app running in the background.

I think when using the app on a Linux smartphone (small screens) it would be good if the focus didn't jump directly to any input fields; perhaps this would also eliminate the behavior when switching between portrait and landscape mode, so that the "layer" one was previously on doesn't change.

---

- Operating System (Linux/Mac/Windows/iOS/Android): Linux Debian Forky (arch64) / Posh /  FLX1
- Delta Chat Version: 2.25.3 (flathub)
- Steps to reproduce the problem:
- Screenshots:

Image 1
<img height="400" alt="Image 1" src="https://github.com/user-attachments/assets/75c183a1-d890-4eab-9c91-1c533f18d738" />

Image 2
<img height="400" alt="Image 2" src="https://github.com/user-attachments/assets/dd36d507-1390-4c67-87fb-7869f6b42571" />

Image 3
<img width="400" alt="Image 3" src="https://github.com/user-attachments/assets/6b44cfc5-e904-46eb-94ac-cadc171b29f2" />

Image 4
<img width="400" alt="Image 4" src="https://github.com/user-attachments/assets/1e1c368f-4e19-415d-b979-6bf47d43a23a" />

Image 5
<img height="400" alt="Image 5" src="https://github.com/user-attachments/assets/433b3da2-f5b1-41fc-8bc2-a6f3462bce64" />

Image 6
<img height="400" alt="Image 6" src="https://github.com/user-attachments/assets/bfe0341a-bc3e-47fc-b923-225c3fe672a2" />

**Comment by WofWca** [comment]
Thanks for the details!

1. I'm not entirely sure what the common practice is, but at the very least for mobile I think it makes sense to remove this behavior. It's anyway easy to focus the search field with Ctrl + F, or otherwise it's just 3 "Tab" presses away from the "default", topmost focusable item.
  For reference, when you open WhatsApp for desktop, the initially focused item is the search input. Do you think WhatsApp's behavior is also bad? Or does it somehow not open the keyboard?
2. Here I'm not really sure if we can remove this behavior entirely for both mobile and Desktop. I think on Desktop it's somewhat expected that if you open the chat then you can start typing the message immediately, as if it's a dialog with `autofocus`.
  This is also how it works on WhatsApp for Desktop. But on Delta Chat Android the keyboard doesn't open.
3. Interesting point. I believe the responsible code is here,

    https://github.com/deltachat/deltachat-desktop/blob/3bc21041bfbafea2da9c297adcdffa28528e8480/packages/frontend/src/hooks/chat/useSelectLastChat.ts#L21-L51

    and I suspect that this behavior is not 100% intentional. Rather it's there only to select the last chat when switching the account or on app startup.
    I think we need to fix it, no doubt.

Related issue: #4590.

**Comment by Simon-Laux** [comment]
> 2. Here I'm not really sure if we can remove this behavior entirely for both mobile and Desktop.

Priority for delta chat desktop is desktop, linux mobile is not a priority, so I would always decide in favor of desktop UX when there is no easy way to make both work.

**Comment by misc-de** [comment]
It will increasingly make no difference under Linux whether I use a desktop or a smartphone – that's the fun part. With surprisingly many apps, this is already the case, and it also applies to DeltaChat :)

> Priority for delta chat desktop is desktop, linux mobile is not a priority, so I would always decide in favor of desktop UX when there is no easy way to make both work.

You have my full understanding if you don’t want to make any big changes to the current code because of that. It already works satisfactorily, from a user’s perspective, it’s always just a small thing that could make the app better for usages on upcoming mobile :P

>     1. I'm not entirely sure what the common practice is, but at the very least for mobile I think it makes sense to remove this behavior. It's anyway easy to focus the search field with Ctrl + F, or otherwise it's just 3 "Tab" presses away from the "default", topmost focusable item.
>        For reference, when you open WhatsApp for desktop, the initially focused item is the search input. Do you think WhatsApp's behavior is also bad? Or does it somehow not open the keyboard?

I haven’t been using the other messengers for a long time, and I’ve switched from Signal to DeltaChat with a few people because I’ve stopped using Google or Apple devices for about 10 years now (As Daily Driver). So, my experience with it is more neutral in this case and less influenced by the behavior I’ve been conditioned to by other apps.

For example, when I switch to a chat because someone sent me something, I often have to deactivate the keyboard first because there’s not enough space on the screen for me to read both pictures and text. The keyboard already takes up at least 1/3 of the screen, and I don’t really need it in that moment.

>     2. Here I'm not really sure if we can remove this behavior entirely for both mobile and Desktop. I think on Desktop it's somewhat expected that if you open the chat then you can start typing the message immediately, as if it's a dialog with `autofocus`.
>        This is also how it works on WhatsApp for Desktop. But on Delta Chat Android the keyboard doesn't open.

On the desktop, I have enough screen space that this behavior doesn’t even stand out :) There, I wouldn’t want to intervene with the current behavior either.

Maybe an app setting (like autofocus) would be useful/possible so the user could actively choose the behavior without the need to differentiate between mobile/desktop in the code.

**Comment by nicodh** [comment]
Since we already have to cover many different OS and Environments it is preferred not to add too many settings for specific use case. And mobile is not a supported use case. Did you see that there is a mobile client for linux https://open-store.io/app/deltatouch.lotharketterer ?
